scratch-debugger/debug.sh

```diff
@@ -0,0 +1,179 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if [[ $# == 0 ]]; then
+  echo >&2 "USAGE: $0 POD_NAME [POD_NAMESPACE CONTAINER_NAME]"
+  exit 1
+fi
+
+# Customizable parameters
+TMP_SUBDIR="${TMP_SUBDIR:-debug-tools}"
+CONTEXT="${KUBECONTEXT:-}"
+DEBUGGER_NAME="${DEBUGGER_NAME:-debugger}"
+ARCH="${ARCH:-amd64}"
+
+# Name & namespace of target pod
+NAME=$1
+NAMESPACE=${2:-}
+CONTAINER_NAME=${3:-}
+
+# Internal variables
+KUBECTL="kubectl"
+[ -z "$CONTEXT" ] || KUBECTL="$KUBECTL --context=$CONTEXT"
```

Comment:
inconsistent use of ${FOO} vs $FOO.  I hate "${FOO}" but it is what most shell in kube uses.